### ISSUE TITLE
Battle maps

### ISSUE THREAD

**Issue by tukkek** [tukkek/javelin#109]
The first file here seems to have some interesting generators that could be useful if adapted as Dungeon maps https://kusigrosz.jimdo.com/roguelike-related/

**Comment by tukkek** [comment]
A Ruins type of map could be a simple generator that creates horizontal and vertical lines on top of a pre-existing map. The size and density of these walls could allow for different types of ruins - like sparse ones in the desert (eroded by time), medium ones in grasslands and dense ones underground.

**Comment by Zireael07** [comment]
The site from OP is down, only accessible via Wayback Machine - I hope you saved the scripts, as I've noticed the machine is bad at serving files...

**Comment by tukkek** [comment]
You are right, @Zireael07 - I found a cached version of it but unfortunately it doesn't hold the ZIP file https://web.archive.org/web/20160516005628/http://kusigrosz.jimdo.com/roguelike-related

There's probably some similar snippets on RogueBasin http://www.roguebasin.com/index.php?title=Articles#Map

**Comment by tukkek** [comment]
This one is an interesting, natural-looking dungeon map http://arcade.academy/examples/procedural_caves_cellular.html

**Comment by tukkek** [comment]
[This multi-reddit](https://www.reddit.com/r/Star_Wars_Maps+battlemaps+dndmaps/) is also a good source of inspiration.

**Comment by tukkek** [comment]
Chasm map for Mountain terrain. References:

* https://www.deviantart.com/gamaweb/art/The-Infernal-Chasm-clean-battlemap-30x30-857390224
* https://twitter.com/CornyMaps/status/1344784713693720579/photo/1

Should be easy enough to have the chasm as black tiles.

The challenge here is to have the chasm as the main tactical feature of the map but still have enough access to all parts of the terrain without flying or ranged attacks - and at the same time not have those access be major chokepoints for the AI. Wide bridges are the obvious initial approach here but they have to balanced in a way that they don't invalidate the chasm itself - probably better to have narrower and more bridges.

To unify the two styles referenced above, an approach would be to have Faults - a single fault would be something like the first image with just a chasm in the middle. More Faults would spread off of the initial one, generating a chasm more like the second image.
